### PR TITLE
Core: Modularize core

### PR DESCRIPTION
**Pull request by orogvany**
The idea here is to break apart the core from UI.

This allows us more options to provide (autogenerated swagger) clients libs,
as well as allow for other UI implementations (as the old swing client is showing its age) while keeping
the core module as it is.

Would also permit server/docker images to not include the UI components.

This makes running unit tests against the core much faster, as the UI tests are fairly slow, so running unit tests takes too long, which leads to devs running it less.

### CODE REVIEW

File: core/src/main/java/org/semux/util/SystemUtil.java [hunk 1]
```diff
@@ -294,13 +294,14 @@ public static void setLocale(Locale locale) {
...
         if (version == null) {
-            try {
-                version = IOUtil.readStreamAsString(SemuxGui.class.getClassLoader().getResourceAsStream("VERSION"))
-                        .trim();
-            } catch (IOException ex) {
-                logger.info("Failed to read version.");
-                version = "unknown";
-            }
+//            try {
+//                //todo
```
Comment (orogvany) [inline core/src/main/java/org/semux/util/SystemUtil.java]: oops, still need to fix this one.

**Comment by semuxgo** [comment]
Can we break down this PR into several smaller pieces?

**Comment by orogvany** [comment]
I'd prefer not to redo it.  Apart from the renames, it's pretty small.  Most of the added lines are just the new poms.

What'd you have in mind?  Just core and gui to start?  We had some requests for client extraction, so seemed like a good time to do it.

**Comment by orogvany** [comment]
Will take second pass at this and just pull out ui

**Pull request closed without merging**